consensus/src/block_storage/block_tree.rs

```diff
@@ -241,6 +259,16 @@ impl BlockTree {
             let linkable_block = LinkableBlock::new(block);
             let arc_block = Arc::clone(linkable_block.executed_block());
             assert!(self.id_to_block.insert(block_id, linkable_block).is_none());
+            // Note: the assumption is that we have/enforce unequivocal proposer election.
+            if let Some(old_block_id) = self.round_to_ids.get(&arc_block.round()) {
+                warn!(
+                    "Multiple blocks received for round {}. Previous block id: {}",
+                    arc_block.round(),
+                    old_block_id
+                );
```

Comment:
this probably should return Error?

Reply:
I just wanted to retain existing behavior. I found that this situation is very possible. If there was a local timeout and then the proposal was received, then we will have duplicate blocks due to the NIL block (before RoundTimeout change that is).